### PR TITLE
Dont overwrite deployment id with self

### PR DESCRIPTION
**Pull request by tobio**
## Description
If the TF config explicitly uses the current deployment id as the observability deployment id we shouldn't rewrite the read value to `self`. 

## Related Issues

Fixes https://github.com/elastic/terraform-provider-ec/issues/782

## How Has This Been Tested?

Unit tests 

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ec/ecresource/deploymentresource/deployment/v2/deployment_read_test.go [hunk 1]
```diff
@@ -1723,3 +1728,100 @@ func Test_getDeploymentTemplateID(t *testing.T) {
 		})
 	}
 }
+
+func Test_ProcessSelfInObservability(t *testing.T) {
+	tests := []struct {
+		name                              string
+		deployment                        *Deployment
+		baseObservability                 *observabilityv1.Observability
```
Comment (dimuon) [inline ec/ecresource/deploymentresource/deployment/v2/deployment_read_test.go:1736]: Can we use `observabilityv2.Observability` instead?

File: ec/ecresource/deploymentresource/deployment/v2/deployment_read_test.go [hunk 2]
```diff
@@ -1723,3 +1728,100 @@ func Test_getDeploymentTemplateID(t *testing.T) {
...
+				},
+			},
+			expectedObservabilityDeploymentID: ec.String("another-deployment-id"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var obj types.Object
+			if tt.baseObservability != nil {
```
Comment (dimuon) [inline ec/ecresource/deploymentresource/deployment/v2/deployment_read_test.go:1805]: Shall we cover the case when `base.Observability` is `Unknown`? (Sorry, if I missed something).

File: ec/ecresource/deploymentresource/deployment/v2/deployment_read.go [hunk 3]
```diff
@@ -270,19 +273,34 @@ func (dep *Deployment) parseCredentials(resources []*models.DeploymentResource)
...
 
 	if dep.Observability.DeploymentId == nil {
-		return
+		return nil
+	}
+
+	var baseObservability v1.ObservabilityTF
+	diags := base.Observability.As(ctx, &baseObservability, basetypes.ObjectAsOptions{
+		UnhandledNullAsEmpty:    true,
+		UnhandledUnknownAsEmpty: true,
```
[the remark this answers is not in the document]
Comment (tobio) [reply ec/ecresource/deploymentresource/deployment/v2/deployment_read.go:288]: Correct. If `base.Observability` is null or unknown then we would retain the existing behaviour, i.e `observability.deployment_id` would be updated to `self` if it equals the deployment id.

File: ec/ecresource/deploymentresource/deployment/v2/deployment_read_test.go (hunk 2 above)
Comment (tobio) [reply ec/ecresource/deploymentresource/deployment/v2/deployment_read_test.go:1831]: Yep, added.

**Review by dimuon — APPROVED** [review]
LGTM (it needs changelog)

**Pull request merged**